neuro_extras/main.py

```diff
@@ -355,99 +363,48 @@ async def _data_cp(
-            elif suffixes[-2:] == [".tar", ".bz2"] or suffixes[-1] in (".tbz2", ".tbz"):
-                command = "tar"
-                args = ["jxvf", str(file), "-C", str(tmp_dst_url)]
-            elif suffixes[-1] == ".tar":
-                command = "tar"
-                args = ["xvf", str(file), "-C", str(tmp_dst_url)]
-            elif suffixes[-1] == ".gz":
-                command = "gunzip"
-                args = [str(file), str(tmp_dst_url) + file.name[:-3]]
-            elif suffixes[-1] == ".zip":
-                command = "unzip"
-                args = [str(file), "-d", str(tmp_dst_url)]
-            else:
-                raise ValueError(
-                    f"Don't know how to extract file {file.name}"
-                    f"Supported archive types: {', '.join(SUPPORTED_ARCHIVE_TYPES)}"
-                )
-
-            click.echo(f"Running {command} {' '.join(args)}")
-            proc = await asyncio.create_subprocess_exec(command, *args)
-            returncode = await proc.wait()
-            if returncode != 0:
-                raise click.ClickException(f"Extraction failed: {proc.returncode}")
-            else:
-                if file.exists():
-                    # gunzip removes src after extraction, while tar - not
-                    file.unlink()
-
+            # download data into subfolder of target for extraction
+            dst_name = origin_dst_name + "/" + origin_src_url.name
         if compress:
-            file = Path(destination_url.path.split("/")[-1])
-            tmp_dst_archive = Path(TEMP_UNPACK_DIR / file)
-            suffixes = file.suffixes
-            if suffixes[-2:] == [".tar", ".gz"] or suffixes[-1] == ".tgz":
-                command = "tar"
-                args = ["zcf", str(tmp_dst_archive), "-C", str(source_url.path), "."]
-            elif suffixes[-2:] == [".tar", ".bz2"] or suffixes[-1] in (".tbz2", ".tbz"):
-                command = "tar"
-                args = ["jcf", str(tmp_dst_archive), str(source_url.path)]
-            elif suffixes[-1] == ".tar":
-                command = "tar"
-                args = ["cf", str(tmp_dst_archive), str(source_url.path)]
-            elif suffixes[-1] == ".gz":
-                command = "gzip"
-                args = ["-r", str(tmp_dst_archive), str(source_url.path)]
-            elif suffixes[-1] == ".zip":
-                command = "zip"
-                args = [str(tmp_dst_archive), str(source_url.path)]
-            else:
-                raise ValueError(
-                    f"Don't know how to compress to archive type {file.name}. "
-                    f"Supported archive types: {', '.join(SUPPORTED_ARCHIVE_TYPES)}"
+            # preserve origin destination name for compressor
+            if origin_dst_name == origin_src_url.name:
+                logging.warning(
+                    "Source file already has required archive extension. "
+                    "Skipping compression step."
```

Comment:
what if `origin_dst_name == origin_src_url.name == "file.txt"` ?

Reply:
We will raise an error "unsupported arcive type" since this branch supposes that we are archiving the data.